spec/defines/dns_logging_channel_spec.rb

```diff
@@ -0,0 +1,248 @@
+            :log_type  => 'file',
+            :file_path => '/path/to/file.log',
+            :file_size => '5m',
+          }
+        end
+
+        it { is_expected.to_not compile }
+      end
+
+      context 'with log_type = syslog and all options set' do
+        let(:params) do
+          {
+            :log_type      => 'file',
+            :file_path     => '/path/to/file.log',
+            :file_size     => '5m',
+            :file_versions => 4,
+          }
+        end
+
+        it 'should have valid channel contents' do
+          verify_concat_fragment_exact_contents(catalogue, "dns_logging_channel_#{title}.dns", [
+            '  channel test {',
+            '    file "/path/to/file.log" versions 4 size 5m;',
+            '  };',
+          ])
+        end
+      end
+
+      context 'with severity set' do
+        let(:params) do
+          {
+            :log_type => 'stderr',
+            :severity => 'dynamic',
+          }
+        end
+
+        it 'should have valid channel contents' do
+          verify_concat_fragment_exact_contents(catalogue, "dns_logging_channel_#{title}.dns", [
+            '  channel test {',
+            '    stderr;',
+            '    severity dynamic;',
+            '  };',
+          ])
+        end
+      end
+
+      context 'with print_category set' do
+        let(:params) do
+          {
+            :log_type       => 'stderr',
+            :print_category => 'yes',
+          }
+        end
+
+        it 'should have valid channel contents' do
+          verify_concat_fragment_exact_contents(catalogue, "dns_logging_channel_#{title}.dns", [
+            '  channel test {',
+            '    stderr;',
+            '    print-category yes;',
+            '  };',
+          ])
+        end
+      end
+
+      context 'with print_severity set' do
+        let(:params) do
+          {
+            :log_type       => 'stderr',
+            :print_severity => 'yes',
+          }
+        end
+
+        it 'should have valid channel contents' do
+          verify_concat_fragment_exact_contents(catalogue, "dns_logging_channel_#{title}.dns", [
+            '  channel test {',
+            '    stderr;',
+            '    print-severity yes;',
+            '  };',
+          ])
+        end
+      end
+
+      context 'with print_time set' do
+        let(:params) do
+          {
+            :log_type   => 'stderr',
+            :print_time => 'yes',
+          }
+        end
+
+        it 'should have valid channel contents' do
+          verify_concat_fragment_exact_contents(catalogue, "dns_logging_channel_#{title}.dns", [
+            '  channel test {',
+            '    stderr;',
+            '    print-time yes;',
+            '  };',
+          ])
+        end
+      end
+
+      context 'fail when order => 1' do
+        let(:params) do
+          {
+            :log_type => 'stderr',
+            :order    => 1,
+          }
+        end
+
+        it { is_expected.to_not compile }
```

Comment:
You can also use this to check that it compiles because order failed it:
```suggestion
        it { is_expected.to compile.and_raise_error(%r{order}) }
```

(same below for order 99)